internal/xds/translator/route.go

```diff
@@ -135,6 +140,31 @@ func buildXdsRouteAction(routeName string) *route.RouteAction {
 	}
 }
 
+func buildXdsWeightedRouteAction(httpRoute *ir.HTTPRoute) *route.RouteAction {
+	invalidBackendWeight := uint32(100 * float64(httpRoute.BackendWeights.Invalid) / float64(httpRoute.BackendWeights.Valid+httpRoute.BackendWeights.Invalid))
```

Comment:
No need those calculation, you can just use the values and total_weights is optional in xDS.